public/app/features/panel/panellinks/link_srv.ts

```diff
@@ -134,30 +134,36 @@ export const getDataFrameVars = (dataFrames: DataFrame[]) => {
   const suggestions: VariableSuggestion[] = [];
   const keys: KeyValue<true> = {};
 
-  for (const frame of dataFrames) {
-    for (const field of frame.fields) {
-      const displayName = getFieldDisplayName(field, frame, dataFrames);
+  if (dataFrames.length !== 1) {
+    // It's not possible to access fields of other dataframes. So if there are multiple dataframes we need to skip these suggestions.
+    // Also return early if there are no dataFrames.
+    return [];
+  }
 
-      if (keys[displayName]) {
-        continue;
-      }
+  const frame = dataFrames[0];
```

Comment:
Just removed the outer loop here as there was no need to loop over an array we know to contain only one element